packages/typescript-checker/src/grouping/node.ts

```diff
@@ -0,0 +1,38 @@
+import { Mutant } from '@stryker-mutator/api/src/core';
+
+export class Node {
```

Comment:
I'm thinking this might not need to be a class. Maybe it can be a simple interface:

```ts
interface Node {
  parents: Set<Node>;
  children: Set<Node>;
}
```

What do you think?

Reply:
In @danny12321 and my opinion this should be a class. The node knows who its parents and children are so i think its easier to do this object oriented instead of a functional way of programming. For instance if you would do it functional the function getAllChildReferencesIncludingSelf would look like
```ts
getAllChildReferencesIncludingSelf(allChildReferences: Set<Node> = new Set<Node>(), nodeToSearchIn: Node){
    allChildReferences.add(nodeToSearchIn);
    nodeToSearchIn.children.forEach((child) => {
      if (!allChildReferences.has(child)) {
        child.getAllChildReferencesIncludingSelf(allChildReferences);
      }
    });
    return allChildReferences;
  }
```
I think using the this keyword is more elegant than nodeToSearchIn